src/cli/add_test.py

```diff
@@ -0,0 +1,72 @@
+"""Integration and function testing for add.py"""
+
+from pathlib import Path
+from unittest.mock import MagicMock, patch
+
+from click.testing import CliRunner
+
+from src.api.wrappers.cf_wrapper import CurseForgeWrapper
+from src.cli.add import INVALID_FILE_ID_ERROR_MSG, Add
+from src.config.lockfile_entry import LockfileEntry
+from src.lib.asset import CurseForgeAsset
+from src.util.enum import AssetType, Platform, Side
+
+
+@patch('src.config.lockfile.LOCKFILE_FILENAME', 'test_m3.lock.json')
+@patch('src.config.config.CONFIG_FILENAME', 'test_m3.json')
+@patch.object(LockfileEntry, 'create_lockfile_entry_from_resp_obj')
+@patch.object(CurseForgeWrapper, 'get_mod')
+@patch.object(CurseForgeWrapper, 'get_asset_files')
+@patch('src.util.asset_management.download_file')
+# pylint: disable-next=too-many-positional-arguments, too-many-arguments, too-many-locals
+def test_add(
+        mock_download_file, mock_get_asset_files, mock_get_mod,
+        mock_create_lockfile_entry_from_resp_obj, copy_test_data_directory,
+        current_dir, tmp_path, create_file, read_json_file):
+    """Tests that the add command installs the given asset to the project."""
+    ref_path = current_dir / 'testdata/'
+    runner = CliRunner()
+
+    mock_get_asset_return = MagicMock()
+    mock_asset_data = MagicMock()
+    mock_asset_data.modId = 71
```

Comment:
what are these magic numbers

If you use constants, declare at the top as `MOD_ID_FOR_TEST`